### PR TITLE
Mschier/ci

### PR DESCRIPTION
**Pull request by marcschier**
Add container builds and sdl tasks to pipeline.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -82,9 +64,118 @@ jobs:
...
+    continueOnError: true
+    condition: succeededOrFailed()
+  - task: PostAnalysis@1
+    displayName: 'Post Analysis'
+    inputs:
+      BinSkim: true
+      CredScan: true
+      RoslynAnalyzers: true
+  - script: |
+      if not "%BUILD_SOURCEBRANCHNAME%" == "master" echo ##vso[task.setvariable variable=MY_BRANCH;]develop
```
Comment (cristipogacean) [inline azure-pipelines.yml]: Do we still want to make use of develop branch?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,22 +1,10 @@
...
-      dockerfile: ./reverse-proxy/Dockerfile
-    ports:
-      - "10080:10080"
-    depends_on:
-      - registryservice
-      - gatewayservice
-      - vaultservice
-      - twinservice
-      - historyservice
   registryservice:
```
Comment (cristipogacean) [inline docker-compose.yml]: Microsoft.Azure.IIoT.Services.OpcUa.Jobs service is missing

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Graph/src/Microsoft.Azure.IIoT.OpcUa.Graph.csproj [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <Description>Azure Industrial IoT OPC UA Graph components</Description>
-    <VersionSuffix>prerelease001</VersionSuffix>
+    <VersionSuffix>$(PrereleaseSuffix)</VersionSuffix>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Gremlin.Net.CosmosDb" Version="0.3.4.4-rc1" />
```
Comment (cristipogacean) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Graph/src/Microsoft.Azure.IIoT.OpcUa.Graph.csproj]: is this RC version ok?

File: modules/Dockerfile.opc-twin [hunk 4]
```diff
@@ -1,17 +1,31 @@
-FROM microsoft/dotnet:2.2-sdk AS build
```
Comment (cristipogacean) [inline modules/Dockerfile.opc-twin]: I assume this file can be deleted and the proper Docker.debug file to be used for debugging

File: modules/opc-publisher/docker/linux/amd64/Dockerfile [hunk 5]
```diff
@@ -1,23 +1,27 @@
...
-
-# start it up
-FROM microsoft/dotnet:${runtime_base_tag} AS runtime
-WORKDIR /app
-COPY --from=build /app/opcpublisher/out ./
+COPY --from=publish /out .
 WORKDIR /appdata
-ENTRYPOINT ["dotnet", "/app/opcpublisher.dll"]
\ No newline at end of file
+ENTRYPOINT ["dotnet", "/app/opcpublisher.dll"]
```
Comment (cristipogacean) [inline modules/opc-publisher/docker/linux/amd64/Dockerfile]: we should rename the publisher project file to something like Microsoft.Azure.IIoT.Modules.OpcUa.Publisher for consistency

File: azure-pipelines.yml (hunk 1 above)
Comment (marcschier) [reply azure-pipelines.yml]: Everything != master == develop, this is just the name of the TSA stamp, and we have 2.

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Graph/src/Microsoft.Azure.IIoT.OpcUa.Graph.csproj (hunk 3 above)
Comment (marcschier) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Graph/src/Microsoft.Azure.IIoT.OpcUa.Graph.csproj]: There is only Rc version of this code until cosmosdb supports bytecode gremlin.

File: modules/Dockerfile.opc-twin (hunk 4 above)
Comment (marcschier) [reply modules/Dockerfile.opc-twin]: Yep.

File: modules/opc-publisher/docker/linux/amd64/Dockerfile (hunk 5 above)
Comment (marcschier) [reply modules/opc-publisher/docker/linux/amd64/Dockerfile]: The project file is already named as such.  Just the output file is not.  I leave this until the overall publisher changes that are coming soon.

File: docker-compose.yml (hunk 2 above)
Comment (marcschier) [reply docker-compose.yml]: Nice catch!!

**Pull request merged**